ipywidgets/widgets/widget.py

```diff
@@ -547,6 +548,13 @@ def add_traits(self, **traits):
                 self.keys.append(name)
                 self.send_state(name)
 
+    @classmethod
+    @functools.lru_cache(None)
```

Comment:
I feel like an unlimited cache size is a bit dangerous. Maybe it's not in practice though?